src/main/java/com/google/api/codegen/config/DiscoveryField.java

```diff
@@ -63,7 +65,7 @@ public Schema getDiscoveryField() {
   public String getSimpleName() {
     String name =
         Strings.isNullOrEmpty(schema.reference()) ? schema.getIdentifier() : schema.reference();
-    return Name.anyCamel(name).toLowerCamel();
+    return Name.fromUnderScoreOrCamel(name).toLowerCamel();
```

Comment:
Why will this name be in either form?

Reply:
Hmm do they always start with a lowercase letter?

Reply:
Empirically, yes. The interface's function originated as a wrapper around com.google.api.tools.framework.model.Field.getSimpleName(), which, afaik, returns a string that starts with a lowercase letter.

Reply:
Maybe we could have a slightly more targeted method `Name.anyLower`.

Reply:
Done.